src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.js';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const likedMessages = (id) => {
+    const newLikedData = chatData.map((chat) => {
+      if (chat.id === id) {
+        return { ...chat, liked: (chat.liked = !chat.liked) };
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(newLikedData);
+  };
+
+  const displayLikes = () => {
```

Comment:
Nice helper function help calculate the like count info. Rather than building up a string of hearts, the README is looking for the _number_ of likes. SO you could initialize `hearts` to 0, and then add 1 for each chat that passed the filter.